x-pack/legacy/plugins/reporting/server/core.ts

```diff
@@ -9,29 +9,37 @@ import { first, mapTo } from 'rxjs/operators';
 import {
   ElasticsearchServiceSetup,
   KibanaRequest,
-  SavedObjectsClient,
   SavedObjectsServiceStart,
   UiSettingsServiceStart,
+  IRouter,
+  SavedObjectsClientContract,
+  BasePath,
 } from 'src/core/server';
+import { SecurityPluginSetup } from '../../../../plugins/security/server';
 import { ReportingPluginSpecOptions } from '../';
 // @ts-ignore no module definition
 import { mirrorPluginStatus } from '../../../server/lib/mirror_plugin_status';
+import { ILicense } from '../../../../plugins/licensing/server';
 import { XPackMainPlugin } from '../../xpack_main/server/xpack_main';
-import { PLUGIN_ID } from '../common/constants';
 import { screenshotsObservableFactory } from '../export_types/common/lib/screenshots';
-import { ServerFacade } from '../server/types';
+import { ServerFacade, ScreenshotsObservableFn } from '../server/types';
 import { ReportingConfig } from './';
 import { HeadlessChromiumDriverFactory } from './browsers/chromium/driver_factory';
-import { checkLicenseFactory, getExportTypesRegistry, LevelLogger } from './lib';
+import { checkLicense, getExportTypesRegistry, LevelLogger } from './lib';
 import { ESQueueInstance } from './lib/create_queue';
 import { EnqueueJobFn } from './lib/enqueue_job';
 import { registerRoutes } from './routes';
-import { ReportingSetupDeps } from './types';
 
-interface ReportingInternalSetup {
+export interface ReportingInternalSetup {
   browserDriverFactory: HeadlessChromiumDriverFactory;
   elasticsearch: ElasticsearchServiceSetup;
+  license$: Rx.Observable<ILicense>;
+  basePath: BasePath['get'];
+  router: IRouter;
+  security: SecurityPluginSetup;
```

Comment:
This should be optional (`security?: SecurityPluginSetup `) in case the security plugin is disabled, yes?

I'm looking at the authorized_user_pre_routing tests, and `get_user.ts` treats the security plugin as optional.